yarn-project/circuits.js/src/hints/build_note_hash_read_request_hints.ts

```diff
@@ -0,0 +1,68 @@
+import { type Tuple } from '@aztec/foundation/serialize';
+
+import {
+  type MAX_NEW_NOTE_HASHES_PER_TX,
+  type MAX_NOTE_HASH_READ_REQUESTS_PER_TX,
+  type NOTE_HASH_TREE_HEIGHT,
+} from '../constants.gen.js';
+import { siloNoteHash } from '../hash/index.js';
+import {
+  type MembershipWitness,
+  type NoteHashContext,
+  NoteHashReadRequestHintsBuilder,
+  type ReadRequestContext,
+} from '../structs/index.js';
+import { countAccumulatedItems, getNonEmptyItems } from '../utils/index.js';
+
+function isValidNoteHashReadRequest(readRequest: ReadRequestContext, noteHash: NoteHashContext) {
+  // TODO(#6122)
+  return (
+    // noteHash.value.equals(readRequest.value) &&
```

Comment:
I see that they won't match because of the siloing issues, but is there no check we could do here in the interim?
Seems like the below loop could assign the wrong note hash to a read request as long as the counters are in the range, but I could be missing something.

Reply:
We are using the siloed value to select the note hashes below:
```ts
 const pendingNoteHash = noteHashMap
      .get(siloedValue.toBigInt())
      ?.find(n => isValidNoteHashReadRequest(readRequest, n.noteHash));
```
Actually, now you point this out, we don't need to compare the values in this function at all! Because the note hashes passed to this function are picked by matching the value.

Reply:
Ah yep, I had missed that `siloedValue` is from the `ReadRequest`! Got it - no worries.